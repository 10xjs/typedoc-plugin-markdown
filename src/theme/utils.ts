import * as fs from 'fs';
import * as Handlebars from 'handlebars';
import * as path from 'path';
import { DeclarationReflection, Reflection } from 'typedoc/dist/lib/models/reflections/index';

export function compileTemplate(template: string, data: {}) {
  const source = templateToString(template);
  const md: HandlebarsTemplateDelegate = Handlebars.compile(source);
  return new Handlebars.SafeString(md(data));
}

export function safeString(data: string) {
  return new Handlebars.SafeString(data);
}

export function getAnchorRef(ref: string) {
    return ref.replace(/_|\/|\.| /g, '-').replace(/"/g, '').replace(/ /g, '-').toLowerCase();
  }

function templateToString(template: string) {
  return fs.readFileSync(path.join(__dirname, `/${template}`)).toString();
}
