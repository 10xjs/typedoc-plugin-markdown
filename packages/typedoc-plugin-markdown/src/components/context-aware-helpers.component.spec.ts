import * as Handlebars from 'handlebars';

import { TestApp } from '../../test/utils';

describe(`[component] ContextAwareHelpersComponent`, () => {
  let testApp: TestApp;
  beforeAll(() => {
    testApp = new TestApp(['comments', 'classes']);
    testApp.bootstrap();
  });

  describe(`'comment' helper`, () => {
    test(`should build @link references'`, () => {
      expect(
        Handlebars.helpers.comment.call(
          testApp.project.findReflectionByName('commentWithDocLinks').comment
            .text,
        ),
      ).toMatchSnapshot();
    });

    test(`should convert symbols brackets to symbol links'`, () => {
      expect(
        Handlebars.helpers.comment.call(
          testApp.project.findReflectionByName('commentsWithSymbolLinks')
            .comment.text,
        ),
      ).toMatchSnapshot();
    });

    test(`should convert comments with includes'`, () => {
      expect(
        Handlebars.helpers.comment.call(
          testApp.project.findReflectionByName('commentsWithIncludes').comment
            .text,
        ),
      ).toMatchSnapshot();
    });
  });
});
