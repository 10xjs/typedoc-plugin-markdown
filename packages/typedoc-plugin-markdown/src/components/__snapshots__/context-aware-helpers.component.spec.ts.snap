// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] ContextAwareHelpersComponent 'comment' helper should build @link references' 1`] = `
"Taken from http://usejsdoc.org/tags-inline-link.html.
"
`;

exports[`[component] ContextAwareHelpersComponent 'comment' helper should convert comments with includes' 1`] = `
"[[include:class-example.md]]

This is an example of handlebars include

[[include:class-example.hbs]]
"
`;

exports[`[component] ContextAwareHelpersComponent 'comment' helper should convert symbols brackets to symbol links' 1`] = `
"- Link to an external reflection: [Greeter](../classes/_classes_.greeter.md)
- Link to an internal reflection: [commentsInReturn](_comments_.md#commentsinreturn)
- Link to an undefined reflection: [[VOID]]
"
`;
